examples/hal_lib/makefile-common.mk

```diff
@@ -0,0 +1,132 @@
+SDCARD_MOUNT_PATH ?= /Volumes/BAREAPP
```

Comment:
Looks like this file was copied from the one in `shared/`, with some changes. I'd like to merge your changes with the original file so that we have just one `makefile-common.mk` file in the repo.

Looks like it's just a matter of removing the `-nostdlib` linker flag. It compiles without warnings/errors if I just make this change on arm-none-eabi-gcc version 10.2.1. For version 9 I need to also change `-std=20` to `-std=2a`, like you did. Which is a good change to support both compiler versions.

Is there a reason you removed the `-Wno-volatile` flag? I don't get any warnings when I put that back.